macos/mac_os_resource_snd.ksy

```diff
@@ -0,0 +1,345 @@
+types:
+  data_format:
+    seq:
+      - id: id
+        type: u2
+        enum: data_type
+      - id: options
+        type: u4
+        doc: contains initialisation options for the SndNewChannel function
+    instances:
+      wave_init_channel_mask:
+        value: 0x07
+        doc: wave table only, Sound Manager 2.0 and earlier
+      wave_init:
+        value: options & wave_init_channel_mask
+        enum: wave_init_option
+        if: id == data_type::wave_table_synth
+      init_pan_mask:
+        -orig-id: initPanMask
+        value: 0x0003
+        doc: mask for right/left pan values
+      pan_init:
+        value: options & init_pan_mask
+        enum: init_option
+      init_stereo_mask:
+        -orig-id: initStereoMask
+        value: 0x00C0
+        doc: mask for mono/stereo values
+      stereo_init:
+        value: options & init_stereo_mask
+        enum: init_option
+      init_comp_mask:
+        -orig-id: initCompMask
+        value: 0xFF00
+        doc: mask for compression IDs
+      comp_init:
+        value: options & init_comp_mask
+        enum: init_option
+  sound_command:
+    seq:
+      - id: is_data_offset
+        type: b1
+      - id: cmd
+        type: b15
+        enum: cmd_type
+      - id: param1
+        type: u2
+      - id: param2
+        type: u4
+    instances:
+      sound_header:
+        pos: param2
+        type: sound_header
+        size-eos: true
+        if: is_data_offset and cmd == cmd_type::buffer_cmd
+  sound_header:
+    seq:
+      - id: sample_ptr
+        -orig-id: samplePtr
+        type: u4
+        doc: pointer to samples (or 0 if samples follow data structure)
+      - id: num_samples
+        -orig-id: length
+        type: u4
+        doc: number of samples
+        if: sound_header_type == sound_header_type::standard
+      - id: num_channels
+        -orig-id: numChannels
+        type: u4
+        doc: number of channels in sample
+        if: sound_header_type == sound_header_type::extended or sound_header_type == sound_header_type::compressed
+      - id: sample_rate
+        -orig-id: sampleRate
+        type: unsigned_fixed_point
+        doc: The rate at which the sample was originally recorded.
+      - id: loop_start
+        -orig-id: loopStart
+        type: u4
+        doc: loop point beginning
+      - id: loop_end
+        -orig-id: loopEnd
+        type: u4
+        doc: loop point ending
+      - id: encode
+        type: u1
+        enum: sound_header_type
+        doc: sample's encoding option
+      - id: midi_note
+        type: u1
+        doc: base frequency of sample, expressed as MIDI note values, 60 is middle C
+        -orig-id: baseFrequency
+      - id: extended_or_compressed
+        type: extended_or_compressed
+        if: sound_header_type == sound_header_type::extended or sound_header_type == sound_header_type::compressed
+      - id: sample_area
+        -orig-id: sampleArea
+        size-eos: true
```

Comment:
I would prefer the variant when the size of `sample_area` is calculated, as in your previous versions of the spec:

https://github.com/kaitai-io/kaitai_struct_formats/blob/9adbc8f49d6000f0023dd8c1b8ed2d3faae37d16/macos/mac_os_resource_snd.ksy#L97

https://github.com/kaitai-io/kaitai_struct_formats/blob/9adbc8f49d6000f0023dd8c1b8ed2d3faae37d16/macos/mac_os_resource_snd.ksy#L137-L140

---

https://github.com/kaitai-io/kaitai_struct_formats/blob/9adbc8f49d6000f0023dd8c1b8ed2d3faae37d16/macos/mac_os_resource_snd.ksy#L141

https://github.com/kaitai-io/kaitai_struct_formats/blob/9adbc8f49d6000f0023dd8c1b8ed2d3faae37d16/macos/mac_os_resource_snd.ksy#L196-L199

You can go with multiple `sample_area`s switched on/off using `if`s, or use the conditional operator _(aka ternary op)_ for joining the expressions:

```suggestion
        size: |
          sound_header_type == sound_header_type::standard ? num_samples : 
          sound_header_type == sound_header_type::extended ? num_frames * num_channels * bits_per_sample / 8 :
          _io.size - _io.pos
```

The last fallback expression (`size: _io.size - _io.pos`) is actually an equivalent of `size-eos: true`, but if think that it's better to have a second field switched using `if` with explicit `size-eos: true` instead, it's OK too.